src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/ConstructedEETypeNode.cs

```diff
@@ -137,11 +137,11 @@ protected override DependencyList ComputeNonRelocationBasedDependencies(NodeFact
 
             if (closestDefType.HasGenericDictionarySlot())
```

Comment:
Pre-existing code that you wrote ;)  It looks like the original intent (https://github.com/nattress/corert/commit/e05cca67d48cf8d4afb167245271157ebc3755eb) was to add a dependency on the type's generic dictionary only if we have the extra dictionary VTable slot. It's morphed over time to be please create a template, which we should do for all generics with a canonical subtype, so it's probably safe to change this yeah. I'll try switching it once I put out all the Array<T> fires.